### PR TITLE
[PD-3687] Allow microsites to specify locked css path

### PR DESCRIPTION
**Pull request by vigowei**
## What's this PR do?
Add `locked_css_path` (an S3 folder) to site configurations admin page, to be used with `lock_css`.

- lock_css is enabled, and locked_css_path is provided, template will load CSS from that path
- lock_css is enabled, locked_css_path is not provided, template will load site specific CSS
- lock_css is disabled, locked_css_path is ignored, template will load default CSS

## Screenshots (if appropriate)

image

## Where should the reviewer start?
please look at the migration first, let me know if the description text should be updated.

## How should this be manually tested?
run microsites locally,
go to admin page, Configurations, update css path for my.jobs 
reload my.jobs, style should change/break

### Separation of Concerns

- [x] Back end "business logic" is implemented in plain python, Django model classes, and Django forms. (not Django views or other frameworks)
- [x] Concerns of UI, api, and model are in separate code units.

### Tests

- [x] Base classes used for python tests don't do too much unnecessary work.

### Migrations

- [x] Migrations can be reversed.
- [x] Current production code would continue to work after this migration is applied.

### CODE REVIEW

File: seo/models.py [hunk 1]
```diff
@@ -1226,6 +1229,8 @@ def get_template(self, template_string):
 
     lock_css = models.BooleanField("Lock CSS", default=False,
                                    help_text='Use hard-coded CSS instead of dynamic')
+    locked_css_path = models.CharField(max_length=255, blank=True,
+                                       help_text='Specify the S3 bucket and folder for your CSS files')
```
Comment (almathew) [inline seo/models.py]: Path and file name, not bucket. They don't get a choice about which bucket.

Comment (vigowei) [reply seo/models.py]: updated.

File: seo/models.py [hunk 2]
```diff
@@ -1059,7 +1059,10 @@ def save(self, *args, **kwargs):
         # Increment the revision number so a new cache key will be used for urls
         # of the seosites linked to this configuration.
         self.revision += 1
-        if self.group: self.group.save()
+        if self.group:
+            self.group.save()
+        if not self.lock_css:
+            self.locked_css_path = ''
```
Comment (koonsies) [inline seo/models.py]: So, if I understand this.. If they set a path, but forget to check the box, it clears the path without an error or any other indicator? I think this is a bit heavy handed. I feel like it's a better approach to let the path be set, but simply check the flag as to whether or not to display it. Someone may want to turn on/off the CSS as they are testing, without having to reset the path each time.

**Comment by almathew** [comment]
We need to talk through some of the implementation details with DS.

File: seo/models.py (hunk 2 above)
Comment (vigowei) [reply seo/models.py]: Thanks for the feedback, this has been addressed. Hopefully it's more intuitive now.

**Comment by tdphillips** [comment]
Where did the migration go?

**Comment by vigowei** [comment]
@tdphillips my bad, I forgot to add it after deleting the old version, just pushed it up.

**Comment by almathew** [comment]
I'm ok with this, although I'll leave it open for @JLMcLaugh or @tdphillips (or @koonsies I suppose) to comment on more.

**Comment by tdphillips** [comment]
Did the DS conversation happen? I don't want to merge this if they haven't signed off on whatever was going to be asked.

**Comment by almathew** [comment]
Yes, sorry, it happened before @vigowei re-opened the PR.
DS has confirmed that they're ok with the specifying only folder name(s).

**Pull request merged**